wildfly/connectors/cassandra/connector-cassandra/src/main/java/org/teiid/resource/adapter/cassandra/CassandraConnectionImpl.java

```diff
@@ -18,117 +18,59 @@
 
 package org.teiid.resource.adapter.cassandra;
 
-import java.util.List;
-import java.util.Set;
-
 import javax.resource.ResourceException;
 
+import org.teiid.cassandra.BaseCassandraConnection;
 import org.teiid.logging.LogConstants;
 import org.teiid.logging.LogManager;
-import org.teiid.resource.spi.BasicConnection;
-import org.teiid.translator.TranslatorException;
-import org.teiid.translator.cassandra.CassandraConnection;
 
 import com.datastax.driver.core.*;
+import org.teiid.translator.cassandra.CassandraConnection;
 
 /**
  * Represents a connection to Cassandra database.
  * */
-public class CassandraConnectionImpl extends BasicConnection implements CassandraConnection{
+public class CassandraConnectionImpl extends BaseCassandraConnection implements CassandraConnection {
     private CassandraManagedConnectionFactory config;
-    private Cluster cluster = null;
-    private Session session = null;
-    private Metadata metadata = null;
-    private VersionNumber version;
+
+    public CassandraConnectionImpl(String keyspace) {
+        super(keyspace);
+    }
 
     public CassandraConnectionImpl(CassandraManagedConnectionFactory config, Metadata metadata) {
+        super(getCluster(config),config.getKeyspace(), metadata);
         this.config = config;
-        this.metadata = metadata;
     }
 
     public CassandraConnectionImpl(CassandraManagedConnectionFactory config) {
+        super(getCluster(config), config.getKeyspace());
         this.config = config;
+    }
 
+    private static Cluster getCluster(CassandraManagedConnectionFactory config) {
         Cluster.Builder builder  = Cluster.builder().addContactPoint(config.getAddress());
 
-        if (this.config.getUsername() != null) {
-            builder.withCredentials(this.config.getUsername(), this.config.getPassword());
+        if (config.getUsername() != null) {
```

Comment:
Looks like this whole class's methods can be moved to base class, then you just need to class definition. Make sure if you move log messages the message text also recreated in the API module.

Reply:
I've done this. Just be sure, you meant referencing CassandraManagedConnectionFactory's object with the CassandraConfiguration interface, right?
Also I couldn't understand the log messages thing. What I understand is that you mean "the same message text need to be there when doing it in the api module in BaseCassandraConnection " but i don't know how to make the managedconnectionfactory class field "UTIL" to be made available. Also, is it necessary?

Reply:
if you moved any log messages to another module, then externalized strings in logs there do not have property file that goes with it. see in the "resources" folder there is i18n.properties file. I am saying part of those that are affected also needs to be moved.